### PR TITLE
Slim jimm model

### PR DESCRIPTION
**Pull request by SimoneDutto**
## Description

feature-brach into v3, with merge conflicts

Fixes _JIRA/GitHub issue number_

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [ ] Covered by unit tests
- [ ] Covered by integration tests

### CODE REVIEW

**Review by alesstimec — APPROVED** [review]

File: internal/jimm/model.go [hunk 1]
```diff
@@ -742,16 +743,101 @@ func (j *JIMM) ModelInfo(ctx context.Context, user *openfga.User, mt names.Model
 	return j.mergeModelInfo(ctx, user, mi, m)
 }
 
+// modelSummariesMap is a safe map to add records concurrently because the access is guarded by a Mutex.
+// The read operations are not guarded because only inserts are done concurrently.
+type modelSummariesMap struct {
+	mu             sync.Mutex
```
Comment (ale8k) [inline internal/jimm/model.go:749]: Can just use sync.Map, no need for mutex @SimoneDutto

File: internal/jimm/watcher.go [hunk 2]
```diff
@@ -343,9 +343,9 @@ func (w *Watcher) watchController(ctx context.Context, ctl *dbmodel.Controller)
 						machines++
 						cores += n
 					}
-					m.Cores = cores
-					m.Machines = machines
-					m.Units = int64(len(v.units))
+					// m.Cores = cores
+					// m.Machines = machines
+					// m.Units = int64(len(v.units))
```
Comment (ale8k) [inline internal/jimm/watcher.go]: This is left commented out?

File: internal/jujuapi/modelmanager.go [hunk 3]
```diff
@@ -71,6 +71,7 @@ type ModelManager interface {
 	ImportModel(ctx context.Context, user *openfga.User, controllerName string, modelTag names.ModelTag, newOwner string) error
 	ModelDefaultsForCloud(ctx context.Context, user *dbmodel.Identity, cloudTag names.CloudTag) (jujuparams.ModelDefaultsResult, error)
 	ModelInfo(ctx context.Context, u *openfga.User, mt names.ModelTag) (*jujuparams.ModelInfo, error)
+	ModelSummaries(ctx context.Context, user *openfga.User, maskingControllerUUID string) (jujuparams.ModelSummaryResults, error)
```
Comment (ale8k) [inline internal/jujuapi/modelmanager.go:74]: This interface is confusing lol (not you)

File: internal/jimm/model.go [hunk 4]
```diff
@@ -742,16 +743,101 @@ func (j *JIMM) ModelInfo(ctx context.Context, user *openfga.User, mt names.Model
...
+	defer m.mu.Unlock()
+	if m.modelSummaries == nil {
+		m.modelSummaries = make(map[string]jujuparams.ModelSummaryResult)
+	}
+	m.modelSummaries[summary.Result.UUID] = summary
+}
+
+// ModelSummaries returns the list of modelsummary the user has access to.
+// It queries the controllers and then merge the info from the JIMM db.
+func (j *JIMM) ModelSummaries(ctx context.Context, user *openfga.User, maskingControllerUUID string) (jujuparams.ModelSummaryResults, error) {
```
Comment (ale8k) [inline internal/jimm/model.go:764]: Probavbly better to just name it ListModelSummaries right?

**Review by ale8k — APPROVED** [review]
some comments... Feel like we can do them in this branch

File: internal/jimm/watcher.go (hunk 2 above)
Comment (SimoneDutto) [reply internal/jimm/watcher.go]: this was then removed all togethere in the commit after, so no worries

File: internal/jimm/model.go (hunk 1 above)
Comment (SimoneDutto) [reply internal/jimm/model.go:749]: https://pkg.go.dev/sync#Map

from the go doc, you use sync.Map for very specific use case. Usually they advice you to use a separate mutex.

Comment (ale8k) [reply internal/jimm/model.go:749]: Ah I see, so if routine A handles keys 1-10, and routine B handled keys 11-20, then sync.Map is preferred. Never noticed the second part really.

**Pull request merged**